component-runtime-manager/src/test/java/org/talend/test/FileOutput.java

```diff
@@ -42,18 +47,36 @@ public class FileOutput implements Serializable {
 
     private final FileService service;
 
-    @ElementListener
-    public void length(final JsonObject data) throws IOException {
+    private List<JsonObject> buffer;
+
+    @BeforeGroup
+    public void init() {
+        buffer = new ArrayList<>();
+    }
+
+    @AfterGroup
+    public void commit(final Collection<JsonObject> records) {
```

Comment:
Is it possible to have in the same Output a method with ElementListener and AfterGroup method with records as an input?

Reply:
Yes, you can manage it manually (elementlistener will receive record by record). Here, we have the shortess one.
```
    @ElementListener
    public void bufferize(final JsonObject object) {
        buffer.add(object);
    }

    @AfterGroup
    public void commit() {
        // saves buffered records at once (bulk)
    }
``` 
or just
 ```
    @AfterGroup
    public void commit(final Collection<Record> records) {
        // saves records
    }
```

Reply:
yes, but in your example you use both at the same time
my question is about using it at the same time

Reply:
oups, missed the signature of the commit method... right